### PR TITLE
improve(benchmarks): better cli parsing

### PR DESCRIPTION
**Pull request by 0xNineteen**
### TL;DR
Improved benchmark filtering with an enum-based approach and added a help menu for available benchmarks.

These changes were pulled from #298 and full credit goes to @Rexicon226 

### What changed?
- Introduced a `Benchmark` enum to strictly type available benchmark options
- Added a `--help` flag that displays all available benchmarks
- Replaced string-based benchmark filtering with enum matching
- Updated error handling to show usage information for invalid benchmark names
- Modified logging messages to be more precise

### How to test?
1. Run `zig build benchmark -- --help` to view available benchmarks
2. Test individual benchmarks: `zig build benchmark -- [benchmark_name]`
3. Run all benchmarks: `zig build benchmark -- all`
4. Verify error handling by attempting to run an invalid benchmark name

### Why make this change?
The enum-based approach provides better type safety and prevents typos when selecting benchmarks. The help menu improves user experience by clearly showing available options, making the benchmark system more user-friendly and maintainable.

### CODE REVIEW

**Comment by 0xNineteen** [comment]
* **#391** <a href="https://app.graphite.dev/github/pr/Syndica/sig/391?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/Syndica/sig/391?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/benchmarks.zig [hunk 1]
```diff
@@ -21,6 +21,46 @@ pub const BenchTimeUnit = enum {
...
+    socket_utils,
+    gossip,
+    sync,
+    ledger,
+    bincode,
+};
+
+fn exitWithUsage() noreturn {
+    var stdout = std.io.getStdOut().writer();
+    stdout.writeAll(
```
Comment (yewman) [inline src/benchmarks.zig:40]: nit: alphabetical order?

**Review by yewman — APPROVED** [review]

**Review by Sobeston — DISMISSED** [review]

**Review by Rexicon226 — APPROVED** [review]

File: src/benchmarks.zig (hunk 1 above)
Comment (0xNineteen) [reply src/benchmarks.zig:40]: meh should be fine

File: src/benchmarks.zig [hunk 2]
```diff
@@ -21,6 +21,46 @@ pub const BenchTimeUnit = enum {
...
+        \\ benchmark name [options]
+        \\
+        \\ Available Benchmarks:
```
[the remark this answers is not in the document]
Comment (0xNineteen) [reply src/benchmarks.zig:43]: great idea - [651cec6](https://github.com/Syndica/sig/pull/391/commits/651cec64a8ca583490e60199180d6c92d5fc4fd5)

**Review by yewman — APPROVED** [review]
lgtm

**Pull request merged**